### PR TITLE
Root improvements and restructuration of the style

### PR DESCRIPTION
**Pull request by ahahl**
The former version of the style has been improved. This includes restructuration of several macros and layouts, plus the implementation of new item types.

### CODE REVIEW

**Comment by smarchand** [comment]
bravo !

**Comment by rmzelle** [comment]
Could you change the style indenting back to 2-spaces per level? (you can use http://formatter.citationstyles.org/, as long all special whitespace characters already have been escaped as e.g. `&#160;`)

File: ens-de-lyon-centre-d-ingenierie-documentaire.csl [hunk 1]
```diff
@@ -11,23 +10,19 @@
...
       <email>stephane.marchand@ens-lyon.fr</email>
     </author>
-    <contributor>
-      <name>Isabelle Colon de Carjaval</name>
-      <email>isabelle.colondecarvajal@ens-lyon.fr</email>
-    </contributor>
+    <author>
+      <name>Anatole Lucet</name>
+      <email>anatole.lucet@ens-lyon.fr</email>
+    </author>
```
Comment (rmzelle) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: @ahahl, any reason you removed Isabelle Colon de Carjaval as a contributor?

Comment (smarchand) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: yes, she was supposed to get involved in this project, but she finally did not participate.

Comment (ahahl) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: Apparently because she did not actually contribute to the style.

**Comment by rmzelle** [comment]
@adam3smith, fine with me now, unless you have any suggestions on the coding.

File: ens-de-lyon-centre-d-ingenierie-documentaire.csl [hunk 2]
```diff
@@ -338,18 +376,47 @@
...
+          </if>
+          <else-if variable="locator" match="none">
+            <label variable="page" form="short" suffix="&#160;"/>
+            <text variable="page"/>
+          </else-if>
+        </choose>
+      </if>
+      <!-- Pour les autres cas -->
+      <else>
+        <choose>
```
Comment (adam3smith) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: this choose/if isn't necessary (if there's no locator, CSL won't print anything anyway)

File: ens-de-lyon-centre-d-ingenierie-documentaire.csl [hunk 3]
```diff
@@ -39,297 +35,339 @@
...
+  <!--_-_-_-_-_-SECTION DEFINITION DES MACROS-_-_-_-_-_-_-->
+  <!-- MACRO AUTEUR, EDITEUR SCIENTIFIQUE (ndbp) -->
   <macro name="author">
     <choose>
-      <if variable="author">
-        <names variable="author">
-          <name form="long" and="text" sort-separator=" " initialize-with="." font-style="normal">
+      <if type="interview">
+        <names variable="interviewer">
+          <name form="long" and="text" delimiter-precedes-last="never" sort-separator=" " initialize-with="." font-style="normal">
```
Comment (adam3smith) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: I know they were in the style before, but pretty much none of the `font-style="normal"` attributes are necessary. Could you just delete them all?

File: ens-de-lyon-centre-d-ingenierie-documentaire.csl [hunk 4]
```diff
@@ -39,297 +35,339 @@
...
     </choose>
   </macro>
-  <macro name="editor">
-    <names variable="editor">
+  <!-- MACRO AUTEUR DU LIVRE (ndbp et biblio)-->
+  <macro name="book_author">
+    <names variable="container-author">
       <name form="long" and="text" delimiter-precedes-last="never" sort-separator=" " initialize-with="." font-style="normal">
-        <name-part name="family" font-variant="small-caps"/>
+        <name-part name="family"/>
```
Comment (adam3smith) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: if you don't need the formatting, you don't need to add a `name-part` condition at all. Just delete this line and the next (i.e. 126 and 127)  and then put a / before the > sign in the line above (125)

File: ens-de-lyon-centre-d-ingenierie-documentaire.csl [hunk 5]
```diff
@@ -39,297 +35,339 @@
...
       </name>
-      <label form="short" prefix="&#160;(" suffix=".)"/>
     </names>
   </macro>
+  <!-- MACRO TRANSLATOR (ndbp et biblio)-->
   <macro name="translator">
     <names variable="translator" prefix=" traduit par ">
       <name form="long" and="text" delimiter-precedes-last="never" sort-separator=" " initialize-with="." font-style="normal">
-        <name-part name="family" font-variant="small-caps"/>
+        <name-part name="family"/>
```
Comment (adam3smith) [inline ens-de-lyon-centre-d-ingenierie-documentaire.csl]: same here.

**Comment by rmzelle** [comment]
@ahahl, @smarchand, I went ahead and made the final few changes requested by @adam3smith, and merged the changes. Thanks!

**Comment by ahahl** [comment]
@rmzelle, @adam3smith, Thanks a lot for your comments and your contribution to the improvement of this style! Best,

**Pull request merged**